### DISCUSSION TITLE
Literal type in get request parameters

### DISCUSSION THREAD

**Discussion by elsimqoz** [fastapi/fastapi#8966] [Questions]
```Python
from typing import Literal, Optional

from fastapi import FastAPI


app = FastAPI()


@app.get('/')
def get_test(
    param_int: Optional[Literal[5]] = None,
    param_bool: Optional[Literal[True]] = None,
    param_str: Optional[Literal['a']] = None,
):
    print(param_int, param_bool, param_str)
```

### Description

I expect that I have created params which can be set only to one value. Let's see on Pydantic:
```Python
class A(BaseModel):
    param_str: Optional[Literal['a']] = None
    param_int: Optional[Literal[5]] = None
    param_bool: Optional[Literal[True]] = None
```
`A(param_str='a')` -- expected ok
`A(param_str='b')` -- expected fail
`A(param_int=5)` -- expected ok
`A(param_int=1)` -- expected fail
`A(param_bool=True)` -- expected ok
`A(param_bool=False)` -- expected fail

After let's try to call endpoint
```bash
INFO:     127.0.0.1:53794 - "GET /?param_str=a HTTP/1.1" 200 OK
INFO:     127.0.0.1:53794 - "GET /?param_str=b HTTP/1.1" 422 Unprocessable Entity
INFO:     127.0.0.1:53796 - "GET /?param_int=5 HTTP/1.1" 422 Unprocessable Entity
INFO:     127.0.0.1:53796 - "GET /?param_int=1 HTTP/1.1" 422 Unprocessable Entity
INFO:     127.0.0.1:53798 - "GET /?param_bool=1 HTTP/1.1" 422 Unprocessable Entity
```

So it works only for strings not for int or bool value, doesn't it? How can I use Literal for bool and int?

### Environment

Ubuntu 18.04
Python 3.8.0
fastapi 0.60.1

**Comment by Mause** [comment]
If you use them as strings, it works - `Optional[Literal['True']]` or `Optional[Literal['5']]`

**Comment by elsimqoz** [comment]
@Mause I got it, but what shall I do if I want to use the original types without conversion? Maybe it is a bug or a feature request?

**Comment by wetgi** [comment]
It's a bug / feature request because pydantic can handle this in e.g. post-requests.

```python
from typing import Literal, Optional
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()

# A Pydantic model
class Item(BaseModel):
    quantity: Literal[5, 10 , 15]

@app.post('/')
def post_test(item: Item
):
    print(item)
```

Here you can just post 5, 10, 15 as int and not a string.

**Comment by Cuahchic** [comment]
I just encountered this today with a get request, I'm not too familiar about triaging GitHub requests but does converting this to a discussion mean there's no plans to implement this?

**Comment by LordOfStrings** [comment]
I did a work around by using the annotated block

```python
from typing import Annotated
from fastapi import FastAPI
from pydantic import BeforeValidator

app = FastAPI()

my_query = Annotated[Literal[1, 2, 3], BeforeValidator(int)]

@app.post('/')
def post_test(query:my_query
):
    print(query)
```

**Reply by AndreMPCosta** [reply]
> I did a work around by using the annotated block
> 
> ```python
> from typing import Annotated
> from fastapi import FastAPI
> from pydantic import BeforeValidator
> 
> app = FastAPI()
> 
> my_query = Annotated[Literal[1, 2, 3], BeforeValidator(int)]
> 
> @app.post('/')
> def post_test(query:my_query
> ):
>     print(query)
> ```

Thanks for the workaround, cheers
